svc/FetchService.js

```diff
@@ -50,44 +72,54 @@ export class FetchService {
      * @returns {Promise<Response>} - Promise which resolves to a Fetch Response.
      */
     async fetch(opts) {
-        let {params, method, contentType, url, autoAbortKey} = opts;
+        let {params, method, headers, url, autoAbortKey} = opts;
         throwIf(!url, 'No url specified in call to fetchService.');
+        warnIf(opts.contentType, 'contentType has been deprecated - please pass a "Content-Type" header instead.');
 
         // 1) Compute / install defaults
         if (!method) {
             method = (params ? 'POST' : 'GET');
         }
 
-        if (!contentType) {
-            contentType = (method === 'POST') ? 'application/x-www-form-urlencoded': 'text/plain';
-        }
-
         if (!url.startsWith('/') && !url.includes('//')) {
             url = XH.baseUrl + url;
         }
 
-        // 2) Prepare merged options
+        // 2) Compute headers
+        if (!(headers instanceof Headers)) {
+            const baseHeaders = {
+                'Content-Type': (method === 'POST') ? 'application/x-www-form-urlencoded': 'text/plain'
+            };
+            if (opts.acceptJson) {
+                baseHeaders['Accept'] = 'application/json';
+                delete opts.acceptJson;
```

Comment:
I know this was pre-existing, but caught my eye. 

Any justification for `acceptJson` as a top-level option, vs. again passing in the header? Especially since we go ahead and mutate the opts provided by the caller, which in general we avoid doing.

Reply:
Good catch - will treat it the same as ContentType and just pass in the header